### PR TITLE
automated code inspect

### PR DESCRIPTION
**Pull request by BO41**
Automated code inspection by android studio. Fixing problems AS can fix automatically

part 3 (last)

### CODE REVIEW

File: opentasks-provider/src/main/java/org/dmfs/provider/tasks/handler/RelationHandler.java [hunk 1]
```diff
@@ -248,13 +248,5 @@ else if (type == Relation.RELTYPE_CHILD)
                 db.update(TaskDatabaseHelper.Tables.TASKS, taskValues, Tasks._ID + "=" + oldValues.getLong(relIdCol), null);
             }
         }
-        else if (type == Relation.RELTYPE_SIBLING)
```
Comment (dmfs) [inline opentasks-provider/src/main/java/org/dmfs/provider/tasks/handler/RelationHandler.java]: It's ok to remove the dead branch, but the comment should be left for reference. It's still an unsolved issues.

Comment (BO41) [reply opentasks-provider/src/main/java/org/dmfs/provider/tasks/handler/RelationHandler.java]: sure, will revert this change :wink:

File: opentasks/src/main/java/org/dmfs/tasks/model/TimeZoneChoicesAdapter.java [hunk 2]
```diff
@@ -196,10 +196,9 @@ private String getGMTOffsetString(long millis)
         long absmillis = (millis < 0) ? -millis : millis;
         int minutes = (int) ((absmillis / (1000 * 60)) % 60);
         int hours = (int) ((absmillis / (1000 * 60 * 60)) % 24);
-        StringBuilder builder = new StringBuilder("(GMT");
-        builder.append((millis >= 0) ? '+' : '-');
-        builder.append(String.format("%02d:%02d", hours, minutes)).append(") ");
-        return builder.toString();
+        String builder = "(GMT" + ((millis >= 0) ? '+' : '-') +
+                String.format("%02d:%02d", hours, minutes) + ") ";
+        return builder;
```
Comment (dmfs) [inline opentasks/src/main/java/org/dmfs/tasks/model/TimeZoneChoicesAdapter.java]: Looking at this now, I think we're probably better off with something like this:
```
return String.format("(GMT%c%02d:%02d) ", millis >= 0 ? '+' : '-', hours, minutes)
```
i.e. just `String.format` the entire thing.

File: opentasks/src/main/java/org/dmfs/tasks/SettingsListFragment.java [hunk 3]
```diff
@@ -524,7 +524,7 @@ public void doneSaveListState()
 
 
     /**
-     * This class is used to display a list of accounts. The list can be modified by the {@link #addAccount(Account)} and {@link #clear()} method. The dialog is
+     * This class is used to display a list of accounts. The list can be modified by the  and  method. The dialog is
```
Comment (dmfs) [inline opentasks/src/main/java/org/dmfs/tasks/SettingsListFragment.java]: The second sentence doesn't make much sense anymore now. Maybe just remove it altogether.

File: opentasks/src/main/java/org/dmfs/tasks/TaskListFragment.java [hunk 4]
```diff
@@ -99,8 +99,8 @@
     private final static AbstractFilter COMPLETED_FILTER = new ConstantFilter(Tasks.IS_CLOSED + "=0");
 
     /**
-     * The group descriptor to use. At present this can be either {@link ByDueDate#GROUP_DESCRIPTOR}, {@link ByCompleted#GROUP_DESCRIPTOR} or
-     * {@link ByList#GROUP_DESCRIPTOR}.
+     * The group descriptor to use. At present this can be either ,  or
```
Comment (dmfs) [inline opentasks/src/main/java/org/dmfs/tasks/TaskListFragment.java]: Same here, The second sentence should be removed.

File: opentasks/src/main/java/org/dmfs/tasks/model/constraints/UpdateAllDay.java [hunk 5]
```diff
@@ -58,11 +58,6 @@ public Boolean apply(ContentSet currentValues, Boolean oldValue, Boolean newValu
                 }
 
             }
-            else if ((newValue == null || !newValue) && oldValue != null && oldValue)
```
Comment (dmfs) [inline opentasks/src/main/java/org/dmfs/tasks/model/constraints/UpdateAllDay.java]: same here, the comment is still a valid issue (maybe not a big one though) and we should keep it for reference/as a reminder.

File: opentasks/src/main/java/org/dmfs/tasks/utils/SearchChildDescriptor.java [hunk 6]
```diff
@@ -58,7 +58,7 @@ public SearchChildDescriptor(String authority, String queryColumn, String[] proj
 
     /**
      * Get a new {@link CursorLoader} and update it's selection arguments with the values in {@code cursor} as defined by {@code selectionColumns} in
-     * {@link #ExpandableChildDescriptor(Uri, String[], String, String, int...)}. Also applies any selection defined by <code>filter</code>.
+     * . Also applies any selection defined by <code>filter</code>.
```
Comment (dmfs) [inline opentasks/src/main/java/org/dmfs/tasks/utils/SearchChildDescriptor.java]: The javadoc is broken here. Not sure what this was supposed to mean. Maybe just remove the entire thing. Better have no documentation than an incorrect one.

File: opentasks/src/main/java/org/dmfs/tasks/model/TimeZoneChoicesAdapter.java [hunk 7]
```diff
@@ -196,10 +196,7 @@ private String getGMTOffsetString(long millis)
         long absmillis = (millis < 0) ? -millis : millis;
         int minutes = (int) ((absmillis / (1000 * 60)) % 60);
         int hours = (int) ((absmillis / (1000 * 60 * 60)) % 24);
-        StringBuilder builder = new StringBuilder("(GMT");
-        builder.append((millis >= 0) ? '+' : '-');
-        builder.append(String.format("%02d:%02d", hours, minutes)).append(") ");
-        return builder.toString();
+        return String.format("(GMT%c%02d:%02d) ", millis >= 0 ? '+' : '-', hours, minutes)
```
Comment (dmfs) [inline opentasks/src/main/java/org/dmfs/tasks/model/TimeZoneChoicesAdapter.java]: I forgot the `;` in my example, sorry. This made the test fail.

Comment (BO41) [reply opentasks/src/main/java/org/dmfs/tasks/model/TimeZoneChoicesAdapter.java]: And I just copy / pasted :innocent:

**Pull request merged**